### PR TITLE
Fix session-file parsing logic

### PR DESCRIPTION
**Pull request by lalitb**
A small fix for #953, and add unit/func tests to validate the fix.

The session file parsing logic doesn't take into consideration the \<newline\> character at end of the file, which leads to file-parsing failure and the file getting repopulated with new session data (SDKUid, first-time-launch ts) every time the app is restarted.

### CODE REVIEW

File: lib/offline/LogSessionDataProvider.cpp [hunk 1]
```diff
@@ -184,8 +184,9 @@ namespace MAT_NS_BEGIN
         contents += toString(sessionFirstTimeLaunch);
         contents += '\n';
         contents += sessionSDKUid;
+        // Valid line ends with newline as per posix specs ( https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206)
         contents += '\n';
-
+
```
Comment (reyang) [inline lib/offline/LogSessionDataProvider.cpp:189]: nit: remove these trailing spaces

File: lib/offline/LogSessionDataProvider.cpp [hunk 2]
```diff
@@ -140,7 +140,7 @@ namespace MAT_NS_BEGIN
         }
         std::vector<std::string> v;
         StringUtils::SplitString(content, '\n', v);
-        if (v.size() != 2) {
+        if (v.size() != 3) {
```
Comment (reyang) [inline lib/offline/LogSessionDataProvider.cpp:143]: Is there a compat concern (e.g. the file was written by a previous version of the sdk, and the reader is new version)?

Comment (lalitb) [reply lib/offline/LogSessionDataProvider.cpp:143]: No, it should be fine as we are only changing the parsing logic in the new sdk. The file was correctly written with older sdk, and newer sdk ensure that it is now read correctly too.

Comment (reyang) [reply lib/offline/LogSessionDataProvider.cpp:143]: Thanks.

**Review by reyang — APPROVED** [review]
LGTM.

**Review by ThomsonTan — APPROVED** [review]
:shipit:

File: tests/unittests/LogSessionDataTests.cpp [hunk 3]
```diff
@@ -54,9 +55,32 @@ TEST(LogSessionDataTests, parse_TwoLinesFirstLaunchTooLarge_ReturnsFalse)
...
+   ASSERT_EQ(sessionSDKUid, "bar");
+}
+
+TEST(LogSessionDataTests, getLogSessionData_ValidInput_SessionDataPersists)
+{
+   TestLogSessionDataProvider logSessionDataProvider1(PathToNonEmptyTestSesFile);
+   logSessionDataProvider1.CreateLogSessionData();
+   auto logSessionData1 = logSessionDataProvider1.GetLogSessionData();
+
+   // Create another provider instance and valiate session data is not re-generated
```
Comment (ThomsonTan) [inline tests/unittests/LogSessionDataTests.cpp:78]:
```suggestion
   // Create another provider instance and validate session data is not re-generated
```

**Pull request merged**